src/data-workspace/inputs/date-input.js

```diff
@@ -0,0 +1,88 @@
+import { useConfig } from '@dhis2/app-runtime'
+import { CalendarInput } from '@dhis2/ui'
+import React from 'react'
+import { useField } from 'react-final-form'
+import { useSetDataValueMutation, useUserInfo } from '../../shared/index.js'
+import styles from './inputs.module.css'
+import { InputPropTypes } from './utils.js'
+
+export const DateInput = ({
+    cocId,
+    deId,
+    disabled,
+    fieldname,
+    form,
+    locked,
+    onFocus,
+    onKeyDown,
+}) => {
+    const {
+        data: {
+            settings: { keyUiLocale },
+        },
+    } = useUserInfo()
```

Comment:
I feel like it might be good to have some protection against the situation where things like this are undefined (e.g. `const {data: userInfo} = useUserInfo(); const keyUiLocale = userInfo?.settings?.keyUiLocale), but this seems to be consistent with how we've done it elsewhere in the app, so I guess we're generally choosing to trust that the backend will reply consistently.